ananse/enhancer_binding.py

```diff
@@ -0,0 +1,497 @@
+                    merged_bam = os.path.join(tmpdir, "merged.bam")
+                    bam_merge(self.list_of_bams, merged_bam, self.ncore)
+
+                shutil.copy2(merged_bam, outfile)
+                shutil.copy2(f"{merged_bam}.bai", f"{outfile}.bai")
+            finally:
+                shutil.rmtree(tmpdir, ignore_errors=True)
+
+
+class ScorePeaks:
+    def __init__(self, bed, bam, verbose=True):
+        self.bam = bam  # one sorted & indexed bam file with reads representing enhancer activity
+        self.bed = bed  # one bed file with putative enhancer binding regions
+        self.verbose = verbose
+
+    def count_reads(self, coverage_file):
+        tmpdir = tempfile.mkdtemp()
+        try:
+            bam_coverage = os.path.join(tmpdir, "bam_coverage")
+            count_reads(self.bam, self.bed, bam_coverage)
+
+            shutil.copy2(bam_coverage, coverage_file)
+        finally:
+            shutil.rmtree(tmpdir, ignore_errors=True)
+
+    @staticmethod
+    def normalize_peaks(bam_coverage, bed_output, dist_func="lognorm_dist", **kwargs):
+        """
+        Fit the bam coverage scores to a distribution
+        """
+        bed = pd.read_csv(bam_coverage, header=None, sep="\t")
+        region = bed[0] + ":" + bed[1].astype(str) + "-" + bed[2].astype(str)
+        score = bed[3]
+
+        # obtain a distribution
+        dist_func = Distributions().set(dist_func)
+        dist = dist_func(score, **kwargs)
+
+        # replace scores with distribution values
+        ascending_dist = np.sort(dist)
+        ascending_scores_index = np.searchsorted(np.sort(score), score)
+        norm_score = np.array([ascending_dist[i] for i in ascending_scores_index])
+
+        logn_score = np.log(norm_score + 1)
+        scaled_score = minmax_scale(logn_score)
+        log10_score = np.log10(norm_score + 1)
+
+        data = {
+            "region": region,  # ex: "chr1:0-200"
+            "score": score,
+            "norm_score": norm_score,
+            "logn_score": logn_score,
+            "scaled_score": scaled_score,
+            "log10_score": log10_score,  # used by the original function
+        }
+        bed = pd.DataFrame(data=data)
+        bed.to_csv(bed_output, sep="\t", index=False)
+
+    def run(self, outfile, dist_func="lognorm_dist", force=False, **kwargs):
+        # save the results as it takes ages to run
+        coverage_file = os.path.join(os.path.dirname(outfile), "raw_scoredpeaks.bed")
+        if force or not os.path.exists(coverage_file):
+            if self.verbose:
+                print("Scoring peaks (slow)")
+            self.count_reads(coverage_file)
+
+        # fit bam read counts to specified distribution
+        if force or not os.path.exists(outfile):
+            self.normalize_peaks(coverage_file, outfile, dist_func=dist_func, **kwargs)
+
+
+class ScoreMotifs:
+    def __init__(self, genome, bed, pfmfile=None, ncore=1, verbose=True):
```

Comment:
Here we may need to discuss how to best do this. GimmeMotifs already scans in chunks internally. Ideally, you would not need to set up a lot of logic to do scanning. If there is something that can be done better, we should change that in gimme. Have you had a look at `gimmemotifs.scanner.scan_regionfile_to_table`?

Reply:
This part was left mostly the same, so I haven't looked into gimme much... yet!

Reply:
`scan_regionfile_to_table` is in, and it reduces the amount of logic considerably! 

I am a little worried that the scan results are returned in memory, but we'll see...